### ISSUE TITLE
Original Gameboy (DMG) version?

### ISSUE THREAD

**Issue by sttng** [LIJI32/GBVideoPlayer2#5]
Would it be possible to create an original Gameboy DMG version ?

**Comment by LIJI32** [comment]
It's possible, but very tricky considering the lower clock rate and reduced timer resolution (The DMG can't accurately trigger a timer interrupt every 114 M-cycles)

**Comment by sttng** [comment]
Yeah I understand this maybe quite a challenge given the color Gameboy has twice the clockspeed of the cpu, double the video ram and tripple the ram. Maybe if you reduce the framerate ?

**Comment by LIJI32** [comment]
RAM and VRAM aren't an issue, I hardly use them. Reducing the frame rate won't help either – GBVP plays video in a volatile way – it actively draws an image in real time as the LCD updates the frame.

**Comment by sttng** [comment]
In that sense the only challenge is the less accurate interrupt trigger on the DMG ? Which means you can't be sure to color the 8 (or 16) pixels correctly in each hblank ?

**Comment by T0biasCZe** [comment]
and if the resolution was lowered ?

**Comment by T0biasCZe** [comment]
@LIJI32 Gameboy has 160x144 screen, which is 20x18 tiles, which is 360tiles total. but gameboy has 256tiles limit. (and it can reuse tiles)
If it was only 20x12 tiles (160x96) it would be only 240tiles which would fit. there would be one filler tile for the borders on top and bottom, which would be 241 tiles total. so it would fit all at once into the screen without the mid scan interupt
example picture:
![SonicGBGreem](https://user-images.githubusercontent.com/44525446/125825775-f404519e-87b9-4395-bd05-ffe5ece225a6.png)

**Comment by T0biasCZe** [comment]
in case of classic 4:3 video it could be stretched to 1:1 aspect ratio and be displayed in 16x16 tiles. but since its already 256tiles so it could fit one tile for the borders it could just use the first tile of the video to fill it
example picture:
![obrazek](https://user-images.githubusercontent.com/44525446/125844922-f650b51b-3d53-48fe-bdf0-3e29d24fa376.png)

**Comment by LIJI32** [comment]
As mentioned above, VRAM (and tile count) aren't an issue, GBVP (both 1 and 2) don't display data using tiles; GBVP2 only uses 2 different tiles. The issue is not having double speed mode on the original Game Boy – the halved CPU speed means we get half the horizontal resolution (it gets stretch though), as well as lose the ability to compress video. Additionally, we lose required timer precision, making the code more wasteful on CPU time as we need to take care of timing things ourselves. It's probably better to display DMG video in a different technique.

**Comment by T0biasCZe** [comment]
> as well as lose the ability to compress video
tho the compress ability wouldnt matter too much since since the video is just 2bit 4color, which is a lot smaller than full color video

and with the half horizontal resolution it would just didnt have as many pixels right ? it would be in full screen anyway?

**Comment by T0biasCZe** [comment]
>as well as lose the ability to compress video

tho the compress ability wouldnt matter too much since since the video is just 2bit 4color, which is a lot smaller than full color video

and with the half horizontal resolution it would just didnt have as many pixels and still be in fullscreen right ?
also isnt it already horizontal halved ? 😂

**Comment by T0biasCZe** [comment]
> the halved CPU speed means we get half the horizontal resolution (it gets stretch though)

@LIJI32 well if its stretched back it doesnt matter since it wouldnt be visible on the 2.2" screen
> as well as lose the ability to compress video. 

The video is only 2bit it will not take a lot of space like the color video so it doesnt need the compression as much

**Comment by T0biasCZe** [comment]
Well i tried the original version of the GBVideoplayer, and it sort of works on the original gameboy...

**Comment by sttng** [comment]
> 

You mean the stock released rom for V1 Player or did you modify the source ?

**Comment by Lucas-mother3** [comment]
Well, I will revive the thread because of this:
https://github.com/Staacks/wifi-game-boy-cartridge/tree/master/gb/stream

It's from the GB Wifi cartridge source code, which is impressive by itself, I don't know much assembly myself, but I think it might be helpful.
(This is posted while I ran GBVideoPlayer2 on Termux)

Addendum: Here's also the color version: https://github.com/Staacks/wifi-game-boy-cartridge/tree/master/gb/color-stream

I don't know if these are compatible with most emulators instead of the already limited set of emulators that can run videos encoded on this program.

Addendum 2: Here's the details about how this guy worked on it: https://there.oughta.be/gta5-for-the-game-boy 
https://youtu.be/pX1opw_gsBs - YouTube video
